SpatialGDK/Source/SpatialGDK/Public/Schema/MigrationDiagnostic.h

```diff
@@ -0,0 +1,150 @@
+
+	MigrationDiagnostic() = default;
+
+	static Worker_CommandRequest CreateMigrationDiagnosticRequest()
+	{
+		// Request information from the worker that has authority over the actor that is blocking the hierarchy from migrating
+		Worker_CommandRequest CommandRequest = {};
+		CommandRequest.component_id = SpatialConstants::MIGRATION_DIAGNOSTIC_COMPONENT_ID;
+		CommandRequest.command_index = SpatialConstants::MIGRATION_DIAGNOSTIC_COMMAND_ID;
+		CommandRequest.schema_type = Schema_CreateCommandRequest();
+
+		return CommandRequest;
+	}
+
+	// Respond with information on the worker that has authority over the actor that is blocking the hierarchy from migrating
+	static Worker_CommandResponse CreateMigrationDiagnosticResponse(USpatialNetDriver* NetDriver, Worker_EntityId EntityId,
+																	AActor* BlockingActor)
+	{
+		check(NetDriver != nullptr);
+		check(NetDriver->Connection != nullptr);
+		check(NetDriver->LockingPolicy != nullptr);
+		check(NetDriver->VirtualWorkerTranslator != nullptr);
+
+		Worker_CommandResponse CommandResponse = {};
+
+		CommandResponse.component_id = SpatialConstants::MIGRATION_DIAGNOSTIC_COMPONENT_ID;
+		CommandResponse.command_index = SpatialConstants::MIGRATION_DIAGNOSTIC_COMMAND_ID;
+		CommandResponse.schema_type = Schema_CreateCommandResponse();
+
+		Schema_Object* ResponseObject = Schema_GetCommandResponseObject(CommandResponse.schema_type);
+		Schema_AddInt64(ResponseObject, SpatialConstants::MIGRATION_DIAGNOSTIC_ENTITY_ID, EntityId);
+		Schema_AddBool(ResponseObject, SpatialConstants::MIGRATION_DIAGNOSTIC_REPLICATES_ID, BlockingActor->GetIsReplicated());
+		Schema_AddBool(ResponseObject, SpatialConstants::MIGRATION_DIAGNOSTIC_HAS_AUTHORITY_ID, BlockingActor->HasAuthority());
+
+		const VirtualWorkerId LocalVirtualWorkerId = NetDriver->VirtualWorkerTranslator->GetLocalVirtualWorkerId();
+
+		Schema_AddInt32(ResponseObject, SpatialConstants::MIGRATION_DIAGNOSTIC_AUTHORITY_WORKER_ID,
+						NetDriver->VirtualWorkerTranslator->GetLocalVirtualWorkerId());
+		Schema_AddBool(ResponseObject, SpatialConstants::MIGRATION_DIAGNOSTIC_LOCKED_ID, NetDriver->LockingPolicy->IsLocked(BlockingActor));
+
+		AActor* NetOwner;
+		VirtualWorkerId NewAuthWorkerId;
+
+		FSpatialLoadBalancingHandler MigrationHandler(NetDriver);
+		FSpatialLoadBalancingHandler::EvaluateActorResult Result =
+			MigrationHandler.EvaluateSingleActor(BlockingActor, NetOwner, NewAuthWorkerId);
+		bool bCanMigrate = Result == FSpatialLoadBalancingHandler::EvaluateActorResult::Migrate;
+
+		Schema_AddBool(ResponseObject, SpatialConstants::MIGRATION_DIAGNOSTIC_EVALUATION_ID,
+					   Result == FSpatialLoadBalancingHandler::EvaluateActorResult::Migrate);
+		Schema_AddInt32(ResponseObject, SpatialConstants::MIGRATION_DIAGNOSTIC_DESTINATION_WORKER_ID, NewAuthWorkerId);
+		AddStringToSchema(ResponseObject, SpatialConstants::MIGRATION_DIAGNOSTIC_OWNER_ID, GetNameSafe(NetOwner));
```

Comment:
IIRC, names are not replicated. You should try to get the owner's EntityId.

Reply:
Changed to owner EntityId.